cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLdapService.java

```diff
@@ -35,29 +33,20 @@
     private ClouderaManagerClientFactory clouderaManagerClientFactory;
 
     @Inject
-    private GrpcUmsClient umsClient;
+    private ClouderaManagerLicenseService licenseService;
 
     public void setupLdap(Stack stack, Cluster cluster, HttpClientConfig clientConfig) throws ApiException {
         LdapConfig ldapConfig = cluster.getLdapConfig();
-        ClouderaManagerResourceApi clouderaManagerResourceApi = clouderaManagerClientFactory.getClouderaManagerResourceApi(stack, cluster, clientConfig);
-
-        // Begin the Cloudera Manager trial only if UMS is not enabled. Otherwise, we'll be using a
-        // license from UMS.
-        String userCrn = stack.getCreator().getUserCrn();
-        if (!umsClient.isUmsUsable(userCrn) || StringUtils.isEmpty(
-                umsClient.getAccountDetails(userCrn, userCrn, Optional.empty()).getClouderaManagerLicenseKey())) {
-            LOGGER.info("Enabling trial license.");
-            clouderaManagerResourceApi.beginTrial();
-        } else {
-            LOGGER.info("UMS detected and license key available, skipping trial license.");
-        }
 
         if (ldapConfig != null) {
+            ApiClient client = clouderaManagerClientFactory.getClient(stack, cluster, clientConfig);
+            licenseService.beginTrialIfNeeded(stack.getCreator(), client);
```

Comment:
Now that we have a license service (good work), we should really pull the license service out of the LdapService -- IMO, it really doesn't belong here.